src/main/java/com/example/teamboolean/apprentidash/ApprentiDashController.java

```diff
@@ -116,49 +114,60 @@ public String recordHour(Model m, Principal p){
-        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss");
-        LocalDateTime now = LocalDateTime.now();
-        boolean endOfDay = false;
-
-        if(buttonRenderHelper().equals("clockIn")) {
-            currentDay.setClockIn(now);
-        }else if(buttonRenderHelper().equals(("lunchIn"))) {
-            currentDay.setLunchStart(now);
-        }else if(buttonRenderHelper().equals("lunchOut")) {
-            currentDay.setLunchEnd(now);
-        }else if(buttonRenderHelper().equals("clockOut")){
-            currentDay.setClockOut(now);
-            endOfDay = true;
-        }
-        currentDay.setUser(userRepository.findByUsername(p.getName()));
-        dayRepository.save(currentDay);
-        if(endOfDay){
-            currentDay = new Day();
+
+        AppUser currentUser = userRepository.findByUsername(p.getName());
+
+        if(buttonRenderHelper(currentUser).equals("clockIn")) {
+            currentUser.getCurrentday().setClockIn(LocalDateTime.now());
+        }else if(buttonRenderHelper(currentUser).equals(("lunchIn"))) {
+            currentUser.getCurrentday().setLunchStart(LocalDateTime.now());
+        }else if(buttonRenderHelper(currentUser).equals("lunchOut")) {
+            currentUser.getCurrentday().setLunchEnd(LocalDateTime.now());
+        }else if(buttonRenderHelper(currentUser).equals("clockOut")){
+            currentUser.getCurrentday().setClockOut(LocalDateTime.now());
         }
-        m.addAttribute("workStatus", buttonRenderHelper());
+
+        currentUser.getCurrentday().setUser(currentUser);
+        dayRepository.save(currentUser.getCurrentday());
+        m.addAttribute("workStatus", buttonRenderHelper(currentUser));
         return "redirect:/recordHour";
     }
 
-    public String buttonRenderHelper(){
-        if(currentDay.getClockIn() == null)
+    public String buttonRenderHelper(AppUser currentUser ){
+        if(currentUser.getCurrentday() == null) {
+            Day day = new Day();
+            currentUser.setCurrentday(day);
+        }
+        if(currentUser.getCurrentday().getClockIn() == null)
             return "clockIn";
-        else if(currentDay.getLunchStart() == null)
+        else if(currentUser.getCurrentday().getLunchStart() == null)
             return "lunchIn";
-        else if(currentDay.getLunchEnd() == null)
+        else if(currentUser.getCurrentday().getLunchEnd() == null)
             return "lunchOut";
-        else if(currentDay.getClockOut() == null)
+        else if(currentUser.getCurrentday().getClockOut() == null)
             return "clockOut";
-        return null;
+        else
+            return "notNewDay";
+    }
+
+    @GetMapping ("/additionalDayRecord")
+    public RedirectView makeDay(Principal p){
+        AppUser currentUser = userRepository.findByUsername(p.getName());
+        currentUser.setCurrentday(null);
+        userRepository.save(currentUser);
+        return new RedirectView("/recordHour");
```

Comment:
Okay, so when a user logs in on a new day it will still show this message, correct? Because it looks like this is the only place where the currentDay gets reset.